### PR TITLE
Added all definitions from linux/ptp-clock.h

### PR DESCRIPTION
**Pull request by davidv1992**
This still has some trouble passing CI due to two unnamed anonymous unions in ptp_perout_request. Any guidance on how to deal with those in a better way?

### CODE REVIEW

File: src/unix/linux_like/linux/mod.rs [hunk 1]
```diff
@@ -804,6 +871,19 @@ s_no_extra_traits! {
...
+    pub union __c_anonymous_ptp_perout_request_1 {
+        pub start: ptp_clock_time,
+        pub phase: ptp_clock_time,
+    }
+
+    #[cfg(libc_union)]
+    pub union __c_anonymous_ptp_perout_request_2 {
+        pub on: ptp_clock_time,
+        pub rsv: [::c_uint; 4],
+    }
```
Comment (folkertdev) [inline src/unix/linux_like/linux/mod.rs:886]: based on

```rust
    #[cfg(libc_union)]
    pub union __c_anonymous_ifr_ifru {
        pub ifru_addr: ::sockaddr,
        pub ifru_dstaddr: ::sockaddr,
```

these need that `#[cfg(libc_union)]` pragma. Then at the use site they need a "default" (highest size and alignment of the variants,  I guess)

```rust
    pub struct ifreq {
        /// interface name, e.g. "en0"
        pub ifr_name: [::c_char; ::IFNAMSIZ],
        #[cfg(libc_union)]
        pub ifr_ifru: __c_anonymous_ifr_ifru,
        #[cfg(not(libc_union))]
        pub ifr_ifru: ::sockaddr,
    }
```

**Comment by JohnTitor** [comment]
Could you resolve the merge conflict and CI failure? Thanks!

**Comment by davidv1992** [comment]
@JohnTitor I would love to, but I don't know how to fix the CI. Essentially, it is failing because the ptp_perout_request struct contains unnamed anonymous enums (so both the enum and the field that is of the type of the enum don't have a name), and I don't know how to tell the CI that properly.

**Comment by JohnTitor** [comment]
You can check around:
https://github.com/rust-lang/libc/blob/d0ce87b9154e8d3b147b4e43ba7d668b082cd7b8/libc-test/build.rs#L3480-L3520

**Comment by Brijeshkrishna** [comment]
https://github.com/rust-lang/libc/blob/3195caf241ff1d2059dc353afb9b97da1400e4b6/libc-test/build.rs#L4180-L4184

try this

```(struct_ == "ptp_perout_request" && field == "anonymous_1")```

**Comment by davidv1992** [comment]
Replaced by #3559

**Pull request closed without merging**